include/core/simulation_control.h

```diff
@@ -308,7 +308,10 @@ class SimulationControl
   /**
    * @brief Check if this is the first assembly of the present iteration.
    * If it indeed is the first assembly, then the first_assembly is set to
-   * false.
+   * false. This function is used when providing the residual to the simulation
+   * control object when using steady-bdf methods. Since the residual must be
+   * provided once per time step (and at the beginning) this function is used to
+   * identify that.
```

Comment:
```suggestion
   * false. This function is used when providing the residual to the simulation
   * control object when using steady-bdf methods. Since the residual must be
   * provided once per time-step (and at the beginning) this function is used to
   * identify that.
```